app/database/manager/index.ts

```diff
@@ -178,13 +180,25 @@ class DatabaseManager {
     * @returns {Promise<void>}
     */
     private initServerDatabase = async (serverUrl: string): Promise<void> => {
-        await this.createServerDatabase({
+        const serverDatabase = await this.createServerDatabase({
             config: {
                 dbName: serverUrl,
                 dbType: DatabaseType.SERVER,
                 serverUrl,
             },
         });
+
+        // Migration for config
+        if (serverDatabase) {
+            const {database, operator} = serverDatabase;
+            const oldConfigList = await querySystemValue(database, SYSTEM_IDENTIFIERS.CONFIG).fetch();
+            if (oldConfigList.length) {
+                const oldConfigModel = oldConfigList[0];
+                const oldConfig = oldConfigModel.value;
+                operator.batchRecords([oldConfigModel.prepareDestroyPermanently()]);
+                storeConfig(serverUrl, oldConfig);
```

Comment:
Just looking at the sequence of steps from line 197 to 199.  Shouldn't we clone the value at line 197 instead of holding a reference to oldConfigModel?

Reply:
Unless I am missing something:
- We get the old model
- We get the value of the old model (a reference to the config object somewhere in memory) and store it in `oldConfig`
- We destroy the model, making accessing to `oldConfigModel` not viable. But since what we have in `oldConfig` is the reference to the config object in memory, it should have not been gotten by the garbage collector.
- We use the oldConfig to add to the new store.

Therefore... it should be save to do it this way. What wouldn't be safe is trying to access `oldConfigModel.value` after the destroy.